x-pack/plugins/task_manager/server/task_pool/task_pool.ts

```diff
@@ -165,7 +165,10 @@ export class TaskPool {
                       message: VERSION_CONFLICT_MESSAGE,
                     })
               )
-              .catch((err) => this.handleFailureOfMarkAsRunning(taskRunner, err));
+              .catch((err) => {
+                this.handleFailureOfMarkAsRunning(taskRunner, err);
+                throw err;
```

Comment:
throwing the error here will cause the entire polling cycle to error, which will correctly emit a task claim failure metric but i'm not sure that's the behavior we're looking for? if we claim 10 tasks and 1 of them is failed to be marked as running, I think we should still run the other 9, which I think throwing this error prevents.

Reply:
I guess if the entire bulk update fails then none of the tasks would get updated anyway?

Reply:
cc @mikecote. If we throw this error here, will it have any other downstream effects?

Reply:
@ymao1 

I explored a bit in the code and couldn't find any downstream effects.

The error thrown here gets caught and handled in this code: https://github.com/elastic/kibana/blob/1344d3b2382cf84108d4f1fe255bd626c0fef523/x-pack/plugins/task_manager/server/polling/task_poller.ts#L69

If there's a way to make the successfully updated tasks run while still reporting the errors in our metrics, that would be great.

Reply:
Can't we just emit an event here?

Reply:
>Can't we just emit an event here?

What would happen if there are requests having errors, calling this code path multiple times? If we emit multiple errors, will there be multiple counts in our metrics of task claims w/ error statuses?

Reply:
I had initially thought about emitting an event [here](https://github.com/elastic/kibana/blob/main/x-pack/plugins/task_manager/server/task_running/task_runner.ts#L509)

Reply:
> What would happen if there are requests having errors, calling this code path multiple times? If we emit multiple errors, will there be multiple counts in our metrics of task claims w/ error statuses?

I didn't get it. You mean something like `runSoon` calling this path, getting errors and calling again? 
I think emitting error per request is expected.
--
btw, I think emitting in the line Alexi linked makes sense (If i am not missing anything)
--
Not sure if throwing here breaks the `promise.all` and stops the rest of the tasks being handled. But if we want to throw an error here we can switch to `promise.allSettled` in order to handle tasks separately.

Reply:
>I think emitting error per request is expected.

From my understanding, emitting an event per failure to update a task will correlate to one task claim cycle failure for each task. So if you fail to update 4/10 tasks, you emit 4 events, and the metrics show 4 task claim cycle failures. Which we only want one record for this task claim cycle? 🤔

Reply:
I meant per runSoon request :) 

I think setting a variable like `hasError=true` in the catch block and emitting an event after the `promise.all` once would work.

Reply:
Gotcha, I think we're all aligned.

Going to the original question about running the tasks that did successfully update.. @doakalexi do you know with the current approach if those tasks still run? or if we need to change some code?

Reply:
I believe with the way it works currently, the bulk update will throw an error and sometimes the bulkUpdate will successfully complete and individual tasks will have an `error` field. When the bulk update throws an error it will be caught and thrown again in the catch that I updated in this PR, but I think this is fine bc there are no tasks that were successfully updated, they all fail. In the second case, those tasks that successfully update will still run and the failed ones with the `error` field are handled in the bulkUpdate code. I am not totally sure we need to change the code, but pls let me know if I am misunderstanding or wrong